mlir/lib/Dialect/Rock/Transforms/AlignTiling.cpp

```diff
@@ -875,15 +972,25 @@ LAGenericRewritePattern::matchAndRewrite(linalg::GenericOp laGeneric,
     }
     Value newOutput = tileReadOp.getDest();
     SmallVector<Value> newInputs;
+    // Only populated if this threadwise_read_into is tracking its validity -
+    // that is, if this is part of an input fusion (where we're worried about
+    // re-applying padding in cases where the generic doesn't preserve 0).
+    SmallVector<Value> newValidityRecords;
     addRegisterReadsForTiledOutput(b, laGeneric, tileReadOp,
-                                   globalCoordsToGenericViews, newInputs);
+                                   globalCoordsToGenericViews, newInputs,
+                                   newValidityRecords);
     // Prevent SSA weirdness from register allocations introduced too late.
     // addRegisterReadsForTiledOutput() may have moved laGeneric into a
     // different block. In this case, SSA is already in good shape.
     if (newOutput.getDefiningOp()->getBlock() == laGeneric->getBlock())
       b.moveBeforeIfNeeded(newOutput.getDefiningOp(), laGeneric);
     reconfigureLAGeneric(b, laGeneric, newInputs, newOutput);
-    b.eraseOp(tileReadOp);
+    Value newValidityRecord =
+        reapplyPaddingIfNeeded(laGeneric, newValidityRecords, tileReadOp, b);
+    if (!newValidityRecord)
+      b.eraseOp(tileReadOp);
+    else
+      b.replaceOp(tileReadOp, newValidityRecord);
```

Comment:
shouldn't this be the ThreadwiseReadIntoOp instead of its validityRecord?

Reply:
Nah, it's replacing an op by a set of `Value`s